R/robustbayesianmetaanalysis.R

```diff
@@ -0,0 +1,2299 @@
+  if (x$type %in% c("Two-sided", "One-sided (mon.)", "One-sided")) {
+    x[["priorOdds"]] <- eval(parse(text = x[["priorOdds"]]))
+    x[["parAlpha"]]  <- .RoBMA_options2priors_clean(x[["parAlpha"]])
+    x[["parAlpha1"]] <-
+      .RoBMA_options2priors_clean(x[["parAlpha1"]])
+    x[["parAlpha2"]] <-
+      .RoBMA_options2priors_clean(x[["parAlpha2"]])
+    x[["parCuts"]]   <- .RoBMA_options2priors_clean(x[["parCuts"]])
+    
+  } else if (x$type == "spike" &&
+             any(names(x) %in% c("parAlpha2"))) {
+    x[["priorOdds"]]   <- eval(parse(text = x[["priorOdds"]]))
+    x[["parLocation"]] <- 1
+    
+  } else{
+    eval_names <-
+      c(
+        "parA",
+        "parB",
+        "parAlpha",
+        "parBeta",
+        "parDF",
+        "parLocation",
+        "parMean",
+        "parScale",
+        "parScale2",
+        "parShape",
+        "priorOdds",
+        "truncationLower",
+        "truncationUpper"
+      )
+    
+    for (n in eval_names) {
+      if (!is.null(x[[n]]))
+        x[[n]] <- eval(parse(text = x[[n]]))
+    }
+    
+  }
+  
+  
+  return(x)
+}
+# table filling functions
+.RoBMA_table_fill_coef      <-
+  function(jasp_table,
+           results_table,
+           add_info,
+           options,
+           individual = FALSE) {
+    # add columns
+    jasp_table$addColumnInfo(name = "terms",
+                             title = "",
+                             type = "string")
+    jasp_table$addColumnInfo(name = "mean",
+                             title = gettext("Mean"),
+                             type = "number")
+    jasp_table$addColumnInfo(name = "median",
+                             title = gettext("Median") ,
+                             type = "number")
+    jasp_table$addColumnInfo(
+      name = "lowerCI",
+      title = gettext("Lower"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$results_CI)
+    )
+    jasp_table$addColumnInfo(
+      name = "upperCI",
+      title = gettext("Upper"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$results_CI)
+    )
+    if (individual) {
+      jasp_table$addColumnInfo(name = "error",
+                               title = gettext("MCMC error"),
+                               type = "number")
+      jasp_table$addColumnInfo(name = "ess",
+                               title = gettext("ESS"),
+                               type = "integer")
+      jasp_table$addColumnInfo(name = "rhat",
+                               title = gettext("Rhat"),
+                               type = "number")
+    }
+    
+    # fill rows
+    if (!is.null(results_table)) {
+      for (i in c(1:nrow(results_table))[rownames(results_table) %in% c("mu", "tau")]) {
+        temp_row <- list(
+          terms    = .RoBMA_coef_names(rownames(results_table)[i], add_info),
+          mean     = results_table$Mean[i],
+          median   = results_table$Median[i],
+          lowerCI  = results_table[i, 3],
```

Comment:
It's not possible to do it like that. The tables are generated with the quantiles as column names because the summary function allows specifying quantiles for the summary (not only lower and upper CI, and it doesn't make sense to have different handling for CI and quantities. I think that rjags and rstan function return tables in similar matter.)

Reply:
But you could construct the names if you know the specified credible %, the same way you do it internally in the package, right? I am not trying to be picky, just thinking about protecting against the future bug when you decide to add a Mode, SE, SD, or what have you to the table output, shifting the indexes of the intervals, having you to go through this code again and fixing the indexes.